docs/kbs_attestation_protocol.md

```diff
@@ -1,220 +1,418 @@
+platform where the KBC is located is in the expected security state**, i.e. if
+it runs in a real and harmless HW-TEE, and the software stack is measured and
+verifiable.
+
+In this document, HW-TEE attestation process is the semantics of the application
+layer, which is defined as a simple, universal and extensible
+"Request-Challenge-Attestation-Response" (RACR) protocol. The temporary
+asymmetric key generated by HW-TEE is used to encrypt the response output
+payload, and the standard [HTTP Cookie](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies)
+mechanism is used to avoid the performance problems caused by multiple
+attestation: it's fairly typical to request for several resources, and then send
+from the same KBC multiple Requests. Having to do the whole attestation dance
+would be time consuming and have a latency impact on the guest.
+
+In order to ensure the ease of use and security completeness of KBS, we will use
+[HTTPS](#http-integration) as the default transport protocol to carry the
+application layer semantics designed in this document. This is because HTTPS
+provides KBC with a means to authenticate KBS identity, which effectively avoids
+malicious attackers from hijacking KBS address to impersonate KBS and deceive
+KBC.
+In order to achieve this, the public key of KBS needs to be transmitted to KBC
+through an effective way, and the specific way of public key distribution is out
+of this document scope.
+
+It should be noted that the confidentiality protection provided by HTTPS alone
+is not enough to meet the security requirements of the KBS protocol.
+Therefore, as mentioned above, the key generated by HW-TEE needs to be used to
+encrypt and protect the confidential data.
+
+# `RCAR` semantics
+
+The semantics of attestation defined by KBS is a simple and extensible four-step
+model, which uses JSON structure to organize information. As follows:
+
+1. **Request**: The KBC sends the initial `Request` payload to the KBS, in order
+   to authenticate itself against the KBS, and eventually request resources.
+2. **Challenge**: After receiving the initial `Request` payload, the KBS
+   responds with the `Challenge` payload. This is how the KBS sends the
+   attestation challenge to the KBC. Together with the attestation challenge,
+   the KBS also sends a session identifier to the KBC, as an HTTP Cookie. This
+   session identifier can be used to skip steps 2 and 3 after a successful
+   attestation.
+3. **Attestation**: The KBC replies to the attestation challenge from step 2
+   with an attestation evidence, in order to prove that its environment (HW-TEE)
+   is safe and reliable. The KBC sends an `Attestation` payload to the KBS, that
+   contains the attestation evidence and the HW-TEE generated, ephemeral public
+   key.
+4. **Response**: Upon succesfull attestation, the KBC can request resources by
+   sending HTTP `GET` requests to resource specific endpoints. The KBS responds
```

Comment:
Could you change the order of these sentences to make this step clearer? I think it would be clearer to state first that the KBS will return a successful `Response` payload iff the KBC's `Attestation` payload was successfully validated.

Reply:
I tried to make it clearer now. Let me know if it's not.